### PR TITLE
Search: Fix attribute defaults

### PR DESCRIPTION
**Pull request by noisysocks**
Fixes https://core.trac.wordpress.org/ticket/50608.

Prior to https://github.com/WordPress/gutenberg/pull/22422, the Search block's `label` and `buttonText` attributes defaulted to `__( 'Search' )`. This meant that adding a Search block to a post or page resulted in `<!-- wp:search /-->` being inserted into the HTML.

In https://github.com/WordPress/gutenberg/pull/22422, the defaults were removed so that Search could use a `block.json` file. The default values were instead implemented using a default block variation.

This, however, meant that Search blocks added prior to https://github.com/WordPress/gutenberg/pull/22422 would render with a blank label and button. This is the bug outlined in the ticket above.

This fix is to default label and buttonText to `__( 'Search' )` in the block's `render_callback`. Note that care must be taken here to permit these attributes to be `''` which hides the label and/or button.

| Before | After |
| - | - |
| Screen Shot 2020-07-08 at 12 23 49 | Screen Shot 2020-07-08 at 12 23 23 |

**To test:**

1. Manually insert `<!-- wp:search /-->` into a post's markup.
2. Test that it renders with a label and button text on the frontend.

### CODE REVIEW

**Comment by noisysocks** [comment]
@ellatrix: This should go in 5.5.

File: packages/block-library/src/search/index.php [hunk 1]
```diff
@@ -15,6 +15,11 @@
 function render_block_core_search( $attributes ) {
 	static $instance_id = 0;
 
+	$attributes = wp_parse_args( $attributes, array(
```
Comment (talldan) [inline packages/block-library/src/search/index.php:18]: Could be worth adding a comment mentioning why these are defaulted in this way.

**Review by talldan — APPROVED** [review]
Looks good, I tested and confirmed it fixes the issue. Code change is small but clear.

Is it worth keeping the variation after this?

**Comment by noisysocks** [comment]
> Is it worth keeping the variation after this?

I did wonder this. We could remove the variation but then would have to change the frontend to display `__( 'Search' )` in the editor when `attributes.label === undefined` and `setAttributes( { label: undefined } )` when `label === __( 'Search' )`. This feels pretty cumbersome to me.

**Pull request merged**